src/gui/editors/AutomationEditor.cpp

```diff
@@ -1119,22 +951,33 @@ inline void AutomationEditor::drawCross( QPainter & p )
 
 
 
-inline void AutomationEditor::drawAutomationPoint( QPainter & p, timeMap::iterator it )
+inline void AutomationEditor::drawAutomationPoint(QPainter & p, timeMap::iterator it)
 {
-	int x = xCoordOfTick( it.key() );
-	int y = yCoordOfLevel( it.value() );
-	const int outerRadius = qBound( 3, ( m_ppb * AutomationPattern::quantization() ) / 576, 5 ); // man, getting this calculation right took forever
-	p.setPen( QPen( vertexColor().lighter( 200 ) ) );
-	p.setBrush( QBrush( vertexColor() ) );
-	p.drawEllipse( x - outerRadius, y - outerRadius, outerRadius * 2, outerRadius * 2 );
+	QMutexLocker m(&m_patternEditorMutex);
+
+	int x = xCoordOfTick(it.key());
+	int y;
+	const int outerRadius = qBound(3, (m_ppb * AutomationPattern::quantization()) / 576, 5); // man, getting this calculation right took forever
```

Comment:
Where does 576 come from? Replace comment with how you came up with that value.

Reply:
That number was already there, kind of laugh at the comment when I read it :laughing: 

No idea how whoever wrote this got that number, would have to hunt through the git history to find the original author.

Reply:
Oh! I didn't realize that wasn't you. lol no worries then

Reply:
Just my guess:
- `AutomationPattern::quantization()` is the quantization unit in ticks
- `m_ppb` is the number of pixels for a bar
- `(m_ppb * AutomationPattern::quantization()) / 576` corresponds to 1/3 of the quantization unit if we're working with the normal 4/4 time signature(whJust my guess:ere 1 bar is 192 ticks)

But this is not important for now because it used to be there.

Reply:
@PhysSong You solved the riddle of the Sphinx :laughing: 

I think you nailed it, I tried to work the math:
1 bar is equal to `192/quantization()` notes, so to get the number of pixels per note we would need `m_ppb * 1 bar / (192/quantization())` or `m_ppb * quantization() / 192`. If we want 1/3 of it, `m_ppb * quantization() / 576`.

I'll add it as a comment for future reference, thanks a lot!

Reply:
Comment was added!